spec/tree_spec.rb

```diff
@@ -1,14 +1,79 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
+  before(:all) do
+    @my_tree = Tree.new
+  end  
+
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(@my_tree.is_a? Tree).to be true
+  end
+
+  it 'should tell if the tree has any apples' do
+    expect(@my_tree.any_apples?).to eq(false)
+  end
+  
+  it 'should age one year on running the age command' do
+    @my_tree.age!
+    expect(@my_tree.age).to eq(1)
+  end
+
+  it 'should add apples on running the add_apples command' do
+    @my_tree.add_apples!
+    expect(@my_tree.any_apples?).to be true
+  end
+
+  it 'should return an apple on running the pick_an_apple command' do
+    expect(@my_tree.pick_an_apple.is_a? Apple).to be true
+  end
+
+  it 'should tell if the tree is dead' do
+    until @my_tree.age > 5
+      @my_tree.age!
+    end
+    expect(@my_tree.dead?).to be true
   end
+
 end
 
-describe 'Fruit' do
+describe Fruit do
+
+  before(:all) do
```

Comment:
In the code I've seen at CMM, `let(:my_fruit) { Fruit.new }` is the standard for this sort of thing.  (Or better yet, `subject { Fruit.new }`.  As I understand it, the main reason to prefer the above over `before(:all)` is test independence: if your test fails, you don't want to be worrying about whether it is failing because of something that happened when your object was touched in a different test.  You want the test to point to the failure, so you want a fresh object.  

Well, then how about `before(:each)`?  `let()` is lazy-evaluated, so it won't be run more often than necessary, unlike the contents of a before(:each) block.